astronomer/providers/google/cloud/triggers/bigquery.py

```diff
@@ -378,3 +383,83 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
                 self.log.exception("Exception occurred while checking for query completion")
                 yield TriggerEvent({"status": "error", "message": str(e)})
                 return
+
+
+class BigQueryTableExistenceTrigger(BaseTrigger):
+    """Initialise the BigQuery Table Existence Trigger with needed parameters"""
+
+    def __init__(
+        self,
+        project_id: str,
+        dataset_id: str,
+        table_id: str,
+        google_cloud_conn_id: str,
+        hook_params: Dict[str, Any],
+        poll_interval: float = 4.0,
+    ):
+        super().__init__()
+        self.dataset_id = dataset_id
+        self.project_id = project_id
+        self.table_id = table_id
+        self.google_cloud_conn_id: str = google_cloud_conn_id
+        self.poll_interval = poll_interval
+        self.hook_params = hook_params
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """Serializes BigQueryTableExistenceTrigger arguments and classpath."""
+        return (
+            "astronomer.providers.google.cloud.triggers.bigquery.BigQueryTableExistenceTrigger",
+            {
+                "dataset_id": self.dataset_id,
+                "project_id": self.project_id,
+                "table_id": self.table_id,
+                "google_cloud_conn_id": self.google_cloud_conn_id,
+                "poll_interval": self.poll_interval,
+                "hook_params": self.hook_params,
+            },
+        )
+
+    def _get_async_hook(self) -> BigQueryTableHookAsync:
+        return BigQueryTableHookAsync(gcp_conn_id=self.google_cloud_conn_id)
```

Comment:
I think we should keep the connection param name `gcp_conn_id` in sensor, operator and hook. Since the sync version (`BigQueryTableExistenceSensor`) has `gcp_conn_id,  bigquery_conn_id` but `bigquery_conn_id` has deprecated

Reply:
Used `gcp_conn_id`